frontend/src/auth/pages/RegisterPage.tsx

```diff
@@ -1,3 +1,32 @@
+import { z } from 'zod'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+
+const SignUpSchema = z.object({
+  email: z.string().email(),
+  password: z.string().min(3).max(20),
+})
+type SignUpSchemaType = z.infer<typeof SignUpSchema>
```

Comment:
also move schemas from component to separate files. for example "schemas/signup.ts"